### PR TITLE
[wip] [v1.x] Onnx export support for slicechannel and box_nms

### PR DESCRIPTION
**Pull request by Zha0q1**
add full support fir skucechannel. Unblocked id_index!=-1 case for box_nms. The behavior will not be the same as mxnet but a warning is added to notify the user of this

### CODE REVIEW

**Review by waytrue17 — APPROVED** [review]
LGTM

File: tests/python-pytest/onnx/test_operators.py [hunk 1]
```diff
@@ -923,3 +923,16 @@ def test_onnx_export_convolution(tmp_path, dtype, shape, num_filter, num_group,
                   **kwargs)
     inputs = [x, w] if no_bias else [x, w, b]
     op_export_test('convolution', M, inputs, tmp_path)
+
+
+@pytest.mark.parametrize('dtype', ['float16', 'float32'])
+@pytest.mark.parametrize('num_outputs', [1, 3, 9])
+@pytest.mark.parametrize('axis', [1, 2, -1, -2])
+@pytest.mark.parametrize('squeeze_axis', [True, False])
```
Comment (josephevans) [inline tests/python-pytest/onnx/test_operators.py:931]: Can you provide a numerical value to squeeze_axis? Does the onnx export function correctly see 'True' when passed a 1?

**Review by josephevans — APPROVED** [review]
Small concern noted, but LGTM!

File: tests/python-pytest/onnx/test_operators.py (hunk 1 above)
Comment (Zha0q1) [reply tests/python-pytest/onnx/test_operators.py]: I added '1' as an equivalent to 'True', but I think technically users can also use other values such as 4, 123, although i doubt they would actually use it that way

**Pull request merged**